project/Scope.rst

```diff
@@ -0,0 +1,23 @@
+Scope of Halium
+===============
+
+Halium's scope primarily consists of 2 parts:
+
+1. Halium's Android is a minimal Android distribution designed to run within an LXC container. It will provide a set of interfaces to let the host of LXC container utilize Android's HAL to communicate with the hardware of Android-based smartphones. These interfaces come in 3 forms:
+
+    1. Native Android interfaces. (e.g. rild's socket, Audio HAL (dlopen'ed directly), (original) WiFi HAL (a standard libhybris wrapper))
+    2. Libhybris compat layer. Usually used to wrap Android C++ interfaces, contains some logic, and has its counter-part compiled in the Android tree. (e.g. HWC2 compat layer)
+    3. Custom interfaces. This will contain more than a compat layer and may have quite deep integration within Android code. (e.g. Ubuntu Touch's camera interfaces, which consists of camera compat layer, custom media compat layer, and a deep modification of Android's libmedia). If possible, we would like to reduce this kind of interface as much as possible to reduce our workload, especially those doing deep modification to Android.
+
+2. Halium's middleware is a set of software/libraries/modules on GNU/Linux which talks to the interfaces in 1. and is distro- & toolkit-agnostic enough. For example, this includes pulseaudio-module-droid, but will not include qtubuntu-camera (a client of the aforementioned camera interface, because it's tied to Qt). Halium might not be upstream of most of the middlewares. Thus, most of the time we'll just maintain the minimal patch(es) that make them work in Halium-based environment.
+
+Halium will primarily be a source-only distribution. Downstream distributions (such as Plasma Mobile or Ubuntu Touch) can place some modification into the Android part (for example, Ubuntu Touch's platform-api, additional kernel cmdline to enable AppArmor, or another initramfs). Then, they can build & distribute Halium as part of the distribution.
+
+Halium will provide the reference rootfs to demonstrate how to integrate Halium into distributions and helps porters in developing their port. It'll contain the common middlewares and enough testing tools to help determine if the port is likely to work in more full-blown distributions or not. A reference repo, which is primarily used to build the reference rootfs, will also be provided. Downstream distributions can use this repo but are not required to.
+
+What is not the scope of Halium
+-------------------------------
+
+Halium does not consists of the following:
+
+- Deployment, initramfs and scripts used in bootup sequence. We understand that each distribution has its own requirement, thus Halium will not enforce it. Halium will provide ones for using in the reference rootfs and for distributions' reference, but distribution isn't required to use them.
```

Comment:
You may want to separate this section out a bit. For example:

* Deployment tools
* initramfs tooling
* Scripts used for the boot sequence (systemd units, init.rc scripts)

We understand that every distribution has its own requirements for these tools, so they are not enforced. We provide reference implementations of each of these which may or may not be used by downstream distributions. For example:

* (links to halium-boot, hybris-boot, initramfs-tools-halium, the works)

Reply:
Please see if the examples I added are OK.

Reply:
Yes, looks good to me.